### PR TITLE
Add MSSQL types

### PR DESCRIPTION
**Pull request by pmantica1**

### CODE REVIEW

File: graphql_compiler/schema_generation/sqlalchemy/scalar_type_mapper.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+# The following quote from https://docs.sqlalchemy.org/en/13/core/type_basics.html
+# explains what makes the all-cap classes particular:
+# "This category of types refers to types that are either part of the SQL standard, or are
+# potentially found within a subset of database backends. Unlike the "generic" types, the SQL
+# standard/multi-vendor types have no guarantee of working on all backends, and will only work
+# on those backends that explicitly support them by name. That is, the type will always emit its
+# exact name in DDL with CREATE TABLE is issued."
+# TODO(pmantica1): Add scalar mapping for the following classes: Interval.
+GENERIC_SQL_CLASS_TO_GRAPHQL_TYPE = {
```
Comment (pmantica1) [inline graphql_compiler/schema_generation/sqlalchemy/scalar_type_mapper.py]: Besides prepending "GENERIC" to this variable name, this is a copy paste.

File: graphql_compiler/schema_generation/sqlalchemy/scalar_type_mapper.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    sqltypes.Time: GraphQLDateTime,
+    sqltypes.TIME: GraphQLDateTime,
+    sqltypes.TIMESTAMP: GraphQLDateTime,
+    sqltypes.Unicode: GraphQLString,
+    sqltypes.UnicodeText: GraphQLString,
+    sqltypes.VARCHAR: GraphQLString,
+}
+
+# We do not currently plan to add a mapping for JSON and Binary objects.
+UNSUPPORTED_GENERIC_SQL_TYPES = frozenset({
```
Comment (pmantica1) [inline graphql_compiler/schema_generation/sqlalchemy/scalar_type_mapper.py]: Same as above.

File: graphql_compiler/schema_generation/sqlalchemy/scalar_type_mapper.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+})
+
+MSSQL_CLASS_TO_GRAPHQL_TYPE = {
+    mssqltypes.BIT: GraphQLBoolean,
+    mssqltypes.DATETIME2: GraphQLDateTime,
+    mssqltypes.REAL: GraphQLFloat,
+    mssqltypes.NTEXT: GraphQLString,
+    mssqltypes.SMALLDATETIME: GraphQLDateTime,
+    mssqltypes.TIME: GraphQLDateTime,
+    mssqltypes.TINYINT: GraphQLInt,
```
Comment (pmantica1) [inline graphql_compiler/schema_generation/sqlalchemy/scalar_type_mapper.py]: Will open an issue marking lack of support for TINYINT.

File: graphql_compiler/schema_generation/sqlalchemy/scalar_type_mapper.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+from graphql.type import GraphQLBoolean, GraphQLFloat, GraphQLID, GraphQLString
+import sqlalchemy.dialects.mssql.base as mssqltypes
+import sqlalchemy.sql.sqltypes as sqltypes
+
+from ...schema import GraphQLDate, GraphQLDateTime, GraphQLDecimal, GraphQLInt
+
+
+# The following quote from https://docs.sqlalchemy.org/en/13/core/type_basics.html
+# explains what makes the all-cap classes particular:
```
Comment (ReaLNeroM) [inline graphql_compiler/schema_generation/sqlalchemy/scalar_type_mapper.py]: Nit: "all caps"

File: graphql_compiler/schema_generation/sqlalchemy/scalar_type_mapper.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    sqltypes.Time: GraphQLDateTime,
+    sqltypes.TIME: GraphQLDateTime,
+    sqltypes.TIMESTAMP: GraphQLDateTime,
+    sqltypes.Unicode: GraphQLString,
+    sqltypes.UnicodeText: GraphQLString,
+    sqltypes.VARCHAR: GraphQLString,
+}
+
+# We do not currently plan to add a mapping for JSON and Binary objects.
+UNSUPPORTED_GENERIC_SQL_TYPES = frozenset({
```
Comment (ReaLNeroM) [inline graphql_compiler/schema_generation/sqlalchemy/scalar_type_mapper.py]: As @obi1kenobi told me in #313, [here](https://github.com/kensho-technologies/graphql-compiler/pull/313#commitcomment-33655951), creating an allowed (assuming GENERIC_SQL_CLASS_TO_GRAPHQL_TYPE means allowed) and disallowed list is sometimes redundant. In this case, can we create a set of allowed types, and disallow everything else?

File: graphql_compiler/schema_generation/sqlalchemy/scalar_type_mapper.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    mssqltypes.REAL: GraphQLFloat,
+    mssqltypes.NTEXT: GraphQLString,
+    mssqltypes.SMALLDATETIME: GraphQLDateTime,
+    mssqltypes.TIME: GraphQLDateTime,
+    mssqltypes.TINYINT: GraphQLInt,
+    mssqltypes.UNIQUEIDENTIFIER: GraphQLID,
+
+}
+
+UNSUPPORTED_MSSQL_TYPES = frozenset({
```
Comment (ReaLNeroM) [inline graphql_compiler/schema_generation/sqlalchemy/scalar_type_mapper.py]: Same as above.

File: graphql_compiler/schema_generation/sqlalchemy/scalar_type_mapper.py (hunk 5 above)
Comment (pmantica1) [reply graphql_compiler/schema_generation/sqlalchemy/scalar_type_mapper.py]: We are doing exactly that. We only have the unsupported types for documentation purposes. It's useful to keep track of the SQL classes for which we don't have a mapping.

File: graphql_compiler/schema_generation/sqlalchemy/scalar_type_mapper.py [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+    #     TIMESTAMP type, which is not supported by SQL Server.  It
+    #     is a read-only datatype that does not support INSERT of values.
+    mssqltypes.TIMESTAMP,
+    mssqltypes.VARBINARY,
+    mssqltypes.XML,
+})
+
+SQL_CLASS_TO_GRAPHQL_TYPE = {}
+SQL_CLASS_TO_GRAPHQL_TYPE.update(GENERIC_SQL_CLASS_TO_GRAPHQL_TYPE)
+SQL_CLASS_TO_GRAPHQL_TYPE.update(MSSQL_CLASS_TO_GRAPHQL_TYPE)
```
Comment (bojanserafimov) [inline graphql_compiler/schema_generation/sqlalchemy/scalar_type_mapper.py]: https://github.com/kensho-technologies/graphql-compiler/blob/macro_system/graphql_compiler/macros/macro_expansion.py#L21 Maybe we should make a common util function for this.

Comment (pmantica1) [reply graphql_compiler/schema_generation/sqlalchemy/scalar_type_mapper.py]: Yea makes sense. I'll create a global_utils file in the top level of the directory.

Comment (pmantica1) [reply graphql_compiler/schema_generation/sqlalchemy/scalar_type_mapper.py]: Actually just noticed that you have this in the macro branch. I would have to make like 2 extra PRs to add this. (Add it into master and remove the function from macro_system). This seems like too much work for this. I'll add a TODO.

Comment (pmantica1) [reply graphql_compiler/schema_generation/sqlalchemy/scalar_type_mapper.py]: Actually just noticed that you have this in the macro branch. I would have to make like 2 extra PRs to add this. (Add it into master and remove the function from macro_system). This seems like too much work for this. I'll add a TODO to use it when it gets merged into master.

**Pull request merged**